lib/rules/no-typos.js

```diff
@@ -54,6 +74,14 @@ module.exports = {
           const propertyName = node.property.name;
           reportErrorIfCasingTypo(node, propertyName);
         }
+      },
+
+      MethodDefinition: function (node) {
+        if (!utils.isES6Component(node.parent.parent) || utils.isReturningJSX(node.parent.parent)) {
```

Comment:
Ahh, I misunderstood the purpose of that util function. I guess we do not need to check for that then.